### PR TITLE
Backedgecutter: add assert for unconditional branch

### PR DESCRIPTION
**Pull request by priyasiddharth**

### CODE REVIEW

File: lib/Transforms/Scalar/BackedgeCutter.cc [hunk 1]
```diff
@@ -47,6 +47,24 @@ char BackedgeCutter::ID = 0;
...
 } // namespace
 
+static void createAssert(BranchInst &TI, Function &F, BasicBlock &dst,
+                         SeaBuiltinsInfo &SBI) {
+  // insert verifier.assert{.not} function
+  auto *assertFn =
+      SBI.mkSeaBuiltinFn(TI.getSuccessor(0) == &dst ? SeaBuiltinsOp::ASSERT_NOT
+                                                    : SeaBuiltinsOp::ASSERT,
+                         *F.getParent());
+  auto ci = CallInst::Create(assertFn, TI.getCondition(), "", &TI);
```
Comment (agurfinkel) [inline lib/Transforms/Scalar/BackedgeCutter.cc:57]: what is condition for unconditional branch?

File: lib/Transforms/Scalar/BackedgeCutter.cc [hunk 2]
```diff
@@ -47,6 +47,24 @@ char BackedgeCutter::ID = 0;
 
 } // namespace
 
+static void createAssert(BranchInst &TI, Function &F, BasicBlock &dst,
+                         SeaBuiltinsInfo &SBI) {
+  // insert verifier.assert{.not} function
+  auto *assertFn =
+      SBI.mkSeaBuiltinFn(TI.getSuccessor(0) == &dst ? SeaBuiltinsOp::ASSERT_NOT
```
Comment (agurfinkel) [inline lib/Transforms/Scalar/BackedgeCutter.cc:54]: not sure this is correct for unconditional branches

File: lib/Transforms/Scalar/BackedgeCutter.cc (hunk 1 above)
Comment (priyasiddharth) [reply lib/Transforms/Scalar/BackedgeCutter.cc:57]: Fixed.

File: lib/Transforms/Scalar/BackedgeCutter.cc (hunk 2 above)
Comment (priyasiddharth) [reply lib/Transforms/Scalar/BackedgeCutter.cc:54]: Fixed.

**Review by agurfinkel — APPROVED** [review]

**Pull request merged**